plugins/out_stackdriver/stackdriver.c

```diff
@@ -696,6 +696,33 @@ static int get_severity_level(severity_t * s, const msgpack_object * o,
     return -1;
 }
 
+static int get_stream(msgpack_object_map map)
+{
+    int i;
+    msgpack_object k;
+    msgpack_object v;
+
+    for (i = 0; i < map.size; i++) {
+        k = map.ptr[i].key;
+        v = map.ptr[i].val;
+
+        if (k.type == MSGPACK_OBJECT_STR &&
+            strncmp(k.via.str.ptr, "stream", k.via.str.size) == 0) {
+            if (strncmp(v.via.str.ptr, "stdout", strlen("stdout")) == 0) {
```

Comment:
we cannot assume that the compiler will optimize the strlen() calls, to avoid unnecessary computing time, please use a predefined variable with its size or a macro, but don't use strlen on that way.

Reply:
ah, also before the strncmp, compare the key length to avoid possible unnecessary string parsing

Reply:
updated